source/application/CLS/CLSAppController.cpp

```diff
@@ -92,7 +112,38 @@ void CLSAppController::onScanEditorCreatedImplementation(AMGenericScanEditor *ed
 {
 	Q_UNUSED(editor)
 
-	AMErrorMon::debug(this, CLS_APPCONTROLLER_INFO_UNIMPLEMENTED_METHOD, "Looks like there is no special implementation for onScanEditorCreated(). ");
+	AMErrorMon::debug(this, CLS_APPCONTROLLER_INFO_UNIMPLEMENTED_METHOD, "Looks like there is no beamline implementation for onScanEditorCreated(). ");
+}
+
+void CLSAppController::onRegionOfInterestAdded(AMRegionOfInterest *region)
+{
+	Q_UNUSED(region)
+
+	AMErrorMon::debug(this, CLS_APPCONTROLLER_INFO_UNIMPLEMENTED_METHOD, "Looks like there is no beamline implementation for onRegionOfInterestAdded(). ");
+}
+
+void CLSAppController::onRegionOfInterestRemoved(AMRegionOfInterest *region)
+{
+	Q_UNUSED(region)
+
+	AMErrorMon::debug(this, CLS_APPCONTROLLER_INFO_UNIMPLEMENTED_METHOD, "Looks like there is no beamline implementation for onRegionOfInterestRemoved(). ");
+}
+
+void CLSAppController::onRegionOfInterestBoundingRangeChanged(AMRegionOfInterest *region)
+{
+	Q_UNUSED(region)
+
+	AMErrorMon::debug(this, CLS_APPCONTROLLER_INFO_UNIMPLEMENTED_METHOD, "Looks like there is no beamline implementation for onRegionOfInterestBoundingRangeChanged(). ");
+}
+
+void CLSAppController::onBeamAvailabilityChanged(bool beamAvailable)
+{
+	if (!beamAvailable && !AMActionRunner3::workflow()->pauseCurrentAction())
```

Comment:
Yes, we were suppose to pause the scan action runner. As #2252, this will make the workflow action runner stuck. I use the workflow action runner here until we found a good solution for this.